mlos_core/mlos_core/optimizers/optimizer.py

```diff
@@ -73,201 +75,182 @@ def space_adapter(self) -> Optional[BaseSpaceAdapter]:
     def register(
         self,
         *,
-        configs: pd.DataFrame,
-        scores: pd.DataFrame,
-        context: Optional[pd.DataFrame] = None,
-        metadata: Optional[pd.DataFrame] = None,
+        observations: Optional[Observations] = None,
+        observation: Optional[Observation] = None,
```

Comment:
```suggestion
        observations: Optional[Union[Observation|Observations]] = None,
```

Proposal: accept `observations` as either a single `Observation` or an `Observations` and automatically convert to `Observations` when you find an `Observation`.

That allows you to:

- drop the `*` from the `register()` signature to simplify it
- drop `register_single()` or else turn it into an private internal implementation detail.